compiler/rustc_hir_analysis/src/collect/resolve_bound_vars.rs

```diff
@@ -686,22 +678,19 @@ impl<'a, 'tcx> Visitor<'tcx> for BoundVarContext<'a, 'tcx> {
             hir::TyKind::Ref(lifetime_ref, ref mt) => {
                 self.visit_lifetime(lifetime_ref);
                 let scope = Scope::ObjectLifetimeDefault {
-                    lifetime: self.map.defs.get(&lifetime_ref.hir_id).cloned(),
+                    lifetime: self.map.defs.get(&lifetime_ref.hir_id.local_id).cloned(),
```

Comment:
Is there any way we can assert that the lifetime's `OwnerId` is the same owner that we're resolving here?

Reply:
And elsewhere throughout this file.

Reply:
I'm possibly just being paranoid, though.

Reply:
I'm not convinced this is useful. We don't track the root `OwnerId`, so we'd need to add it, for limited benefit IMHO.